File: LES_SERIE_S1_IA_FS_ALGO_PYTHON_C/Série_de_TD-TP_3/exercice04.c
#include <stdio.h>
int N,somme;
int main(){
    somme=0;
    printf("saisir un entier: ");
    scanf("%d",&N);
    for(int i=1;i<N;i++){
        if(N%i==0){
            somme=somme+i;
        }
    }
    if(N==somme)
        printf("le nombre %d est parfait.\n",N);
    else
        printf("le nombre %d pas parfait.\n",N);
    return 0;
}

File: LES_SERIE_S1_IA_FS_ALGO_PYTHON_C/Série_de_TD-TP_3/exercice04.py
N=int(input("saisir un entier N: "))
somme=0
for i in range(1,N):
    if N%i==0:
        somme=somme+i
if somme==N:
    print(f"le nombre {N} est parfait.")
else:
    print(f"le nombre {N} pas parfait.")